Prepare release 0.9.0 (#362)
probe-rs/probe-rs: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.9.0]

### Added

- Added initial support for FTDI based probes.
- Added support for the STM32L5 family.
- Added support for the STM32G4 family.

> - Added support for ITM tracing over SWO for CMSIS-DAP and J-Link probes.

Reviewer: ```suggestion
- Added support for ITM tracing over SWO in general and drivers for all probes.
```
